### PR TITLE
add chain switching to walletconnect

### LINKED ISSUE #394: AddEthereumChainParameter argument for WalletConnect
WalletConnect wallets are starting to support multiple chains (Polygon, Optimism, Arbitrum, etc), but it looks there isn't support  for adding / switching to those networks in the WalletConnect connector.

E.g. when I try and request `wallet_switchEthereumChain` to Optimism via the WalletConnectProvider it successfully executes in the Rainbow / MetaMask wallets, but throws an error `Invalid chainId 10` after switching and does not update web3-react state.

Hoping this is a pretty easy fix, the logic would look similar to this: https://github.com/NoahZinsmeister/web3-react/blob/main/packages/metamask/src/index.ts#L134-L150

### PR DESCRIPTION
**Pull request by NoahZinsmeister**
closes #394

### CODE REVIEW

**Review by earthtojake — COMMENTED** [review]
When I run this locally with Rainbow wallet the switch prompt doesn't work - might need some more investigation

File: packages/walletconnect/src/index.ts [hunk 1]
```diff
@@ -79,45 +93,69 @@ export class WalletConnect extends Connector {
...
+      }
+
+      // if we're here, we can try to switch networks
+      const desiredChainIdHex = `0x${desiredChainId.toString(16)}`
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      return this.provider!.request<void>({
+        method: 'wallet_switchEthereumChain',
+        params: [{ chainId: desiredChainIdHex }],
+      })
```
[the remark this answers is not in the document]
Comment (NoahZinsmeister) [reply packages/walletconnect/src/index.ts:136]: i played around fairly extensively with `wallet_addEthereumChain` locally - didn't work at all in metamask or rainbow, and introduced loads of edge cases, as unfortunately the request usually doesn't reject, and i really don't want to introduce timeouts 🙈.

i think until there's a walletconnect wallet that uses this feature well and/or there's more standardization around rejecting unsupported chain requests, i'd prefer to stick to just `desiredChainId`

**Comment by NoahZinsmeister** [comment]
hm i can't replicate that problem with rainbow, do you have a specific repro i could try out?

**Comment by earthtojake** [comment]
Oh yeah this is working for me now - tested on the switching improvement branch, think it was my local dev setup

File: packages/walletconnect/src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (earthtojake) [reply packages/walletconnect/src/index.ts]: sgtm!

**Pull request merged**